File: src/main/kotlin/net/yttravelers/spotshare/entities/Tourism.kt
package net.yttravelers.spotshare.entities

import java.time.LocalDateTime
import javax.persistence.*

@Entity
@Table(name = "tourism")
class Tourism {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    var tourismId: Int? = null

    @Column(nullable = false)
    var tourismName: String = ""

    //TODO: 国マスタ
    @ManyToOne
    @JoinColumn(name = "countryCode")
    var tourismCountry: Country? = null

    //TODO: 町マスタ
//    var tourismCity: City

    var tourismOpenTime: LocalDateTime? = null

    var tourismCloseTime: LocalDateTime? = null

    var tourismSummary: String = ""

    var tourismAddress: String = ""

    var tourismUrl: String = ""

}

File: src/main/angular/src/app/model/tourism.ts
import { ICountry } from "./country";

/**
 * 観光地
 */
export interface ITourism {
  /** 観光地ID */
  tourismId?: number;
  /**  観光地名称 */
  tourismName?: string;
  /** 国 */
  country?: ICountry;
  /** 営業開始時間 */
  tourismOpenTime?: string;
  /** 営業終了時間 */
  tourismCloseTime?: string;
  /** 観光地概要 */
  tourismSummary?: string;
  /** 住所 */
  tourismAddress?: string;
  /** URL */
  tourismUrl?: string;
}